### ISSUE TITLE
Install options to discard some files

### ISSUE THREAD

**Issue by yamashi** [xmake-io/xmake#3460]
### Is your feature request related to a problem? Please describe.

I have a pretty large project, and everytime I run xmake install it takes a very long time because it copies all the include files (60k files) that are not needed.

### Describe the solution you'd like

Options for xmake install to discard includes/libs/bins.

**Comment by waruqi** [comment]
```
add_headerfiles("include/**.h|exclude/*.h")
```

use `...|exclude pattern` to set exclude pattern. and `add_installfiles` support it too.

or 

```
remove_headerfiles
```

**Comment by SirLynix** [comment]
sometimes you just want to install (or just update) binaries, for example with android development headers are installed (even though I just want the binaries).

Also headerfiles are used for project generations

**Comment by waruqi** [comment]
> sometimes you just want to install (or just update) binaries, for example with android development headers are installed (even though I just want the binaries).
> 
> Also headerfiles are used for project generations

add_headerfiles("", {install = false})

**Comment by SirLynix** [comment]
yes the point is sometimes you want to have headers, sometimes you don't, and it does not depends on the project configuration (xmake.lua) but on what you want to do.

For example, I sometimes install my project in a folder in order to have the binaries and their dependencies so I can ship them. but sometimes I want to install it system-wide, the very same project.

So I agree here it would be nice to have a way to install only binaries at least

**Comment by waruqi** [comment]
> > sometimes you just want to install (or just update) binaries, for example with android development headers are installed (even though I just want the binaries).
> > Also headerfiles are used for project generations
> 
> add_headerfiles("", {install = false})


sometimes you just want to install (or just update) binaries, for example with android development headers are installed (even though I just want the binaries).

Also headerfiles are used for project generations

```
option("xxx")
add_headerfiles("", {install = get_config("xxx")})
```

```
xmake f --xxx=y
xmake install
```

**Comment by SirLynix** [comment]
I already pointed out the main issue with xmake config (xmake f), it's that you have to set all of your configurations because xmake doesn't support incremental config.

so you'd have to remember to do `xmake f --tests=y --usepch=y --assimp=y --ffmpeg=y --embed_plugins=y --install_headers=n`, not very practical.

Also, xmake install also copies lib files (such as .lib and more), you can't disable these.

A simple `xmake install --bin` to only copy the bin folder on Windows or .so/.dynlib on others would be a nice addition imo.

**Comment by waruqi** [comment]
> I already pointed out the main issue with xmake config (xmake f), it's that you have to set all of your configurations because xmake doesn't support incremental config.
> 
> so you'd have to remember to do , not very practical.`xmake f --tests=y --usepch=y --assimp=y --ffmpeg=y --embed_plugins=y --install_headers=n`
> 
> Also, xmake install also copies lib files (such as .lib and more), you can't disable these.
> 
> A simple to only copy the bin folder on Windows or .so/.dynlib on others would be a nice addition imo.`xmake install --bin`

you can add `on_install` to do it.

**Comment by SirLynix** [comment]
I'm not talking about just copying the final binary produced by a target but also all of its dependencies, something `xmake install` already does and is not easy to reimplement.

**Comment by Wzshun** [comment]
我也同样希望有一个选项，可以仅安装二进制文件，而不是部署库模式。
虽然可以通过每个库去设置`add_headerfiles("", {install = false})`，去移除不必要的头文件，但在大工程里，每个库都要去这样设置显然是有点多余。此外lib文件夹的部署，似乎没有选项去控制。

所以还是希望形如`xmake install --bin`，这样的入参，仅部署可执行文件和其依赖的库。

此外，还希望xmake能够支持增量式拷贝。如`windows的robocopy` 和 `linux的rsync`，在部分工程可能需要拷贝多个大资源时，部署需要花费不少时间。
